src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -31,23 +31,25 @@ public class PngEncoder {
     private final PngEncoderPhysicalPixelDimensions physicalPixelDimensions;
 
     private final boolean usePredictorEncoding;
+    private final boolean tryIndexedEncoding;
 
     private PngEncoder(BufferedImage bufferedImage, int compressionLevel, boolean multiThreadedCompressionEnabled,
             PngEncoderSrgbRenderingIntent srgbRenderingIntent,
-            PngEncoderPhysicalPixelDimensions physicalPixelDimensions, boolean usePredictorEncoding) {
+            PngEncoderPhysicalPixelDimensions physicalPixelDimensions, boolean usePredictorEncoding, boolean tryIndexedEncoding) {
         this.bufferedImage = bufferedImage;
         this.compressionLevel = PngEncoderVerificationUtil.verifyCompressionLevel(compressionLevel);
         this.multiThreadedCompressionEnabled = multiThreadedCompressionEnabled;
         this.srgbRenderingIntent = srgbRenderingIntent;
         this.physicalPixelDimensions = physicalPixelDimensions;
         this.usePredictorEncoding = usePredictorEncoding;
+        this.tryIndexedEncoding = tryIndexedEncoding;
     }
 
     /**
      * Constructs an empty PngEncoder. Usually combined with methods named with*.
      */
     public PngEncoder() {
-        this(null, DEFAULT_COMPRESSION_LEVEL, true, null, null, false);
+        this(null, DEFAULT_COMPRESSION_LEVEL, true, null, null, false, false);
```

Comment:
Performance: Perhaps the default could read the input. If it's an indexed image, the output is also indexed. This should be the most performant way, right?

Reply:
At the moment indexed images will always be encoded indexed. I.e. the flag is only used for images not of type `BufferedImage.TYPE_BYTE_INDEXED`

Reply:
Oh, that leaves the user with no choice at all :( What if I want to _force_ the output to be ARGB, i.e. for faster future conversion to argb?

Reply:
We can scope this out of this PR though! Like a future feature.